jsapp/js/components/common/koboDropdown.es6

```diff
@@ -0,0 +1,202 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import autoBind from 'react-autobind';
+import {bem} from 'js/bem';
+import {KEY_CODES} from 'js/constants';
+import './koboDropdown.scss';
+
+export const KOBO_DROPDOWN_THEMES = {};
+new Set([
+  'light',
+  'dark',
+]).forEach((codename) => {KOBO_DROPDOWN_THEMES[codename] = codename;});
+Object.freeze(KOBO_DROPDOWN_THEMES);
+
+export const KOBO_DROPDOWN_PLACEMENTS = {};
+new Set([
+  'up-left',
+  'up-center',
+  'up-right',
+  'down-left',
+  'down-center',
+  'down-right',
+]).forEach((codename) => {KOBO_DROPDOWN_PLACEMENTS[codename] = codename;});
+Object.freeze(KOBO_DROPDOWN_PLACEMENTS);
+
+bem.KoboDropdown = bem.create('kobo-dropdown');
+bem.KoboDropdown__trigger = bem.KoboDropdown.__('trigger', 'button');
+bem.KoboDropdown__menu = bem.KoboDropdown.__('menu', 'menu');
+bem.KoboDropdown__menuButton = bem.KoboDropdown.__('menu-button', 'button');
+
+/**
+ * A generic dropdown component that accepts any content inside the menu and
+ * inside the opener.
+ *
+ * NOTE: If you need a select-type dropdown, please use `react-select`.
+ *
+ * You can use some existing content elements:
+ * - bem.KoboDropdown__menuButton - a generic dropdown row button
+ * - bem.KoboDropdown__triggerButton - a generic dropdown row button
+ *
+ * @prop {string} [theme] - `light` by default - one of KOBO_DROPDOWN_THEMES
```

Comment:
Curious, why not use the jsdoc syntax for specifying default values: `param {string} [somebody=John Doe]`?

Reply:
Didn't know about the default value! Changed :)